src/ansys/openapi/common/_session.py

```diff
@@ -261,11 +286,52 @@ def with_autologon(self: Api_Client_Factory) -> Api_Client_Factory:
                 return self
         raise ConnectionError("Unable to connect with autologon.")
 
-    def with_oidc(
+    @require_oidc
+    def with_oidc_client_credentials_flow(
```

Comment:
I feel like it would make more sense to take the token_url here, and leave the api_url as the actual service_url. We could always fall back to the api_url as the token_url if it isn't specified, but it just seems a bit clumsy right now.

Reply:
We should get all the information we need in the initial 401 response from the API url and from the well known endpoint at the identity provider. I'll check this branch out and have a look at what it's doing today